core/base/src/TObject.cxx

```diff
@@ -778,6 +779,36 @@ void TObject::SavePrimitiveConstructor(std::ostream &out, TClass *cl, const char
    out << variable_name << " = new " << cl->GetName() << "(" << constructor_agrs << ");" << std::endl;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Save array in the output stream "out".
+/// Create unique variable name based on prefix value
+
+TString TObject::SavePrimitiveArray(std::ostream &out, const char *prefix, Int_t len, Double_t *arr)
```

Comment:
Why is this a `TObject` method? What `TObject` member field does it need?

Reply:
It is solely used for implementation of override of `TObject::SavePrimitive` (And as such is marked as `protected`, a free function would be, for better or worth, more accessible .. here it is left as an implementation detail of sort)

Reply:
As Philippe comment, free function has no big sense while saving array only should be used in `SavePrimitive` implementations. Like in several places of TGraph-derived classes.